File: examples/python/http_get.py
"""HTTP GET — demonstrates host networking with urllib."""

import urllib.request

url = "http://httpbin.org/get"
print(f"GET {url}")

with urllib.request.urlopen(url, timeout=10) as resp:
    print(f"Status: {resp.status}")
    print(f"Content-Type: {resp.headers['Content-Type']}")
    body = resp.read().decode()
    # Print first 500 chars to keep output manageable.
    print(body[:500])

print("HTTP GET test done.")

File: examples/node/http_get.js
// HTTP GET — demonstrates host networking.

const http = require('http');

const url = 'http://httpbin.org/get';
console.log(`GET ${url}`);

http.get(url, { timeout: 10000 }, (res) => {
    let body = '';
    res.on('data', chunk => body += chunk);
    res.on('end', () => {
        console.log(`Status: ${res.statusCode}`);
        console.log(`Content-Type: ${res.headers['content-type']}`);
        console.log(body.slice(0, 500));
    });
}).on('error', (e) => {
    console.error(`Request failed: ${e.message}`);
});
